tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1240,6 +1260,62 @@ port::StatusOr<CudnnRnnParamsDescriptor> CudnnRnnParamsDescriptor::Create(
         (type == 0 ? weights : biases).push_back(region);
       }
     }
+    int hidden_size_v;
+    int num_layers_v;
+    cudnnDropoutDescriptor_t dropout_desc;
+    cudnnRNNInputMode_t input_mode;
+    cudnnDirectionMode_t direction;
+    cudnnRNNMode_t mode;
+    cudnnRNNAlgo_t algo;
+    cudnnDataType_t data_dype;
+    RETURN_IF_CUDNN_ERROR(cudnnGetRNNDescriptor(
+        /*handle=*/cudnn.handle(), /*rnnDesc=*/rnn_desc,
+        /*hiddenSize=*/&hidden_size_v,
+        /*numLayers=*/&num_layers_v,
+        /*dropoutDesc=*/&dropout_desc,
+        /*inputMode=*/&input_mode,
+        /*direction=*/&direction,
+        /*mode=*/&mode,
+        /*algo=*/&algo,
+        /*dataType=*/&data_type));
+    int rec_proj_size_v;
+    int out_proj_size_v;
+#if CUDNN_VERSION >= 7101
+    RETURN_IF_CUDNN_ERROR(cudnnGetRNNProjectionLayers(
+        /*handle=*/cudnn.handle(),
+        /*rnnDesc=*/rnn_desc,
+        /*recProjSize*/ &rec_proj_size_v,
+        /*outProjSize*/ &out_proj_size_v));
+#else
+    return port::Status(port::error::INVALID_ARGUMENT,
```

Comment:
So this will break existing code? Can we use `CUDNN_VERSION >= 7101` to guard this whole newly added code block?

Also, this function is already too long, can we add a util function for the new code?